straxen/plugins/event_patternfit.py

```diff
@@ -0,0 +1,410 @@
+import strax
+import straxen
+import numpy as np
+from scipy.special import betainc, gammaln, loggamma
+import warnings
+export, __all__ = strax.exporter()
+
+
+@export
+@strax.takes_config(
+    strax.Option('s1_optical_map', help="S1 (x, y, z) optical/pattern map.",
+                 default="XENONnT_s1_xyz_patterns_LCE_corrected_qes_MCva43fa9b_wires.pkl"),
+    strax.Option('s2_optical_map', help="S2 (x, y) optical/pattern map.",
+                 default="XENONnT_s2_xy_patterns_LCE_corrected_qes_MCva43fa9b_wires.pkl"),
+    strax.Option('s1_aft_map', help="Date drive S1 area fraction top map.",
+                 default='s1_aft_dd_xyz_XENONnT_Kr83m_41500eV_25May2021.json'),
+    strax.Option('MeanPEperPhoton', help="Mean of full VUV single photon responce",
+                 default=1.2),
+    strax.Option('gain_model',
+                 help='PMT gain model. Specify as (model_type, model_config)'),
+    strax.Option('n_tpc_pmts', type=int,
+                 help='Number of TPC PMTs'),
+    strax.Option('n_top_pmts', type=int,
+                 help='Number of top TPC PMTs'),
+    strax.Option('s1_min_reconstruction_area',
+                 help='Skip reconstruction if S1 area (PE) is less than this'),
+    strax.Option('s2_min_reconstruction_area',
+                 help='Skip reconstruction if S2 area (PE) is less than this'),
+    strax.Option('StorePerChannel', default=False, type=bool,
+                 help="Store normalized LLH per channel for each peak"),
+    strax.Option("max_r", default = straxen.tpc_r, type=float,
```

Comment:
max_r might be a bit too generic as this could mean many different things

Reply:
changed to `max_r_patternfit`, probably it is a bit more clear this way.